drivers/serial/uart_realtek_rts5912.c

```diff
@@ -50,11 +131,66 @@ static int rts5912_uart_init(const struct device *dev)
 		return rc;
 	}
 
+#if defined(CONFIG_PM)
+	const uint32_t uart_reg = uart_ns16550_get_port(dev_cfg->uart_dev);
+	uint32_t iir = sys_read32(uart_reg + REG_IIR) & 0x0f;
+
+	while (iir != IIR_NIP) {
```

Comment:
This loop is unbounded.  Please add a timeout check.

Can you also add a comment about why the polling of the IIR register is needed during the UART initialization.